docs/mermaid_diagrams.rst

```diff
@@ -0,0 +1,207 @@
+.. _mermaid.js: https://mermaid-js.github.io/mermaid/
+
+Mermaid diagrams
+================
+
+.. note::
+    Use of this feature has no affect or affiliation with sphinx's graphviz implementation.
+
+The mkdocs-material theme is equipped to make use of diagrams generated (during page load time)
+with `mermaid.js`_. Although, its implementation relies on a markdown extension that does not get
+used by this sphinx-immaterial theme. Thus, the sphinx-immaterial theme provides with an optional
+directive that exposes the underlying implementation in mkdocs-material theme.
+
+.. confval:: md-mermaid
+
+    The `md-mermaid` directive does support the ``:class:`` and ``:name:`` options which can used
+    as respective class and id specifiers in custom CSS.
+
+    This theme comes with CSS styling that conforms to the chosen primary & accent colors
+    (based on the selected scheme).
+
+    .. md-admonition::
+        :class: missing
+    
+        While all `mermaid.js`_ features should work out-of-the-box, this theme will currently only
+        adjust the fonts and colors for `flowcharts`_, `sequence diagrams <sequence-diagrams>`,
+        `class diagrams <class-diagrams>`, `state diagrams <state-diagrams>`, and
+        `entity-relationship diagrams <entity-relationship-diagrams>`.
+
+Using flowcharts
+----------------
+
+.. code-block:: rst
+
+    .. md-mermaid::
```

Comment:
Note: it might be nice to add a directive for us in our documentation that avoids having to duplicate the code for samples like this.

Reply:
The `sphinx-design` docs use external files to store the duplicated snippets. The external files are also used in test runs via `pytest`. Their docs are a bit more laborious because they also support the myRST parser (which involves duplicated markdown versions of rst snippets).

Reply:
Yeah it was just a suggestion --- I didn't imagine it as an extension provided by this theme to users, but perhaps just something defined for our own documentation, like `confval`.  But it is true it is widely useful for other sphinx extensions, though maybe some customization is needed for different uses.

Reply:
After giving this more thought, IDK if an external extension is practical for everyone's needs (multiple example snippets can rendered identically). I'll try to keep the idea on my radar, but I doubt I'll ever implement something like this.